modules/installation-default-aws-config-yaml.adoc

```diff
@@ -0,0 +1,82 @@
+// Module included in the following assemblies:
+//
+// * installing-aws/installing-customizations-cloud.adoc
+
+[id='installation-default-aws-config-yaml-{context}']
+= Default `install-config.yaml` file for AWS
+
+The default `install-config.yaml` file defines only the information that is
+required to deploy your {product-title} cluster. The installation program
```

Comment:
This is not strictly true.  For example, see [here][1] and surrounding discussion where we decided to include some properties in the wizard-generated `install-config.yaml` for which we have internal defaults.  But the generated `install-config.yaml` certainly defers some configuration to later logic, so I dunno how much you want to discuss vs. glossing over at this stage.  But I thought I'd point it out so you could make an informed decision, whichever way you go.

[1]: https://github.com/openshift/installer/pull/902#discussion_r246956646

Reply:
Well. The description needs to be accurate here. If there's more nuance to the parameters that this file presents and what users can reasonably be expected to do with it, then I need to explain it somewhere. I need more info, though.

Which of the parameters that are in this "default" file but not set by the installation program can a user modify?

Reply:
> Which of the parameters that are in this "default" file but not set by the installation program can a user modify?

Users can modify everything in the `install-config.yaml`, and if they choose invalid values (e.g. configuring zero masters), we'll exit with a validation error.  But here are the three classes of parameters:

* Given in the wizard-created file, and...
    * The installer has no default.  For example, `baseDomain` or the cluster name.  The wizard will prompt you for these, but if you supply your own `install-config.yaml` we'll error out if they aren't populated.
    * The installer can generate a default.  For example, machine-pool `replicas`.  These are in the wizard-created file because it's convenient for the implementation, and there didn't seem to be any harm in that approach (more details in the discussion linked from my earlier comment in this thread).  If you want, you can supply your own `install-config.yaml` without them, and the installer will happily inject its built-in defaults.
* Not given in a wizard-created file.  For example, the `rootVolume` machine-pool configuration on AWS.  The installer is able to generate defaults for these, but it does so in an asset that consumes the install-config, and not in the install-config asset itself.